lib/railway_ipc/rabbit_mq/rabbit_mq_adapter.ex

```diff
@@ -12,6 +12,7 @@ defmodule RailwayIpc.RabbitMQ.RabbitMQAdapter do
 
   def connect do
     Telemetry.track_connecting_to_rabbit(fn ->
+      # credo:disable-for-next-line Credo.Check.Refactor.WithClauses
```

Comment:
Not sure about this one, so disabling the check for now.

Reply:
Yeah, a `with` with only one is weird.
To make it a case, you could do:
```elixir
case Connection.open(connection_url()) do
  {:ok, connection} when not is_nil(connection) ->
    {{:ok, connection}, %{connection: connection}}
  error ->
    {{:error, error}, %{error: error}}
end
```

Reply:
Ok, I'll give that try.